aten/src/ATen/native/TensorIterator.cpp

```diff
@@ -69,6 +69,20 @@ compute_result_type(at::ArrayRef<OperandInfo> operands, const F& predicate) {
   return std::make_tuple(result_type, backend);
 }
 
+static bool needs_cast(const Tensor& tensor, const Type& dst_type) {
+  if (!tensor.defined() || dst_type == tensor.type()) {
+    return false;
+  }
+  if (dst_type.backend() == Backend::CUDA &&
```

Comment:
do you actually mean for these to be backends or do you mean for them to be device types?  Types have `device_type` now.